src/com/boxboat/jenkins/pipeline/BoxBase.groovy

```diff
@@ -247,12 +260,13 @@ abstract class BoxBase<T extends CommonConfigBase> implements Serializable {
     }
 
     def writeTriggers() {
-        if (gitRepo.branch == config.defaultBranch && gitRepo.isBranchTip()) {
+        def triggers = this.triggers()
+        if (triggers != null && gitRepo.isBranchTip()) {
```

Comment:
Will this if block ever execute? `triggers()` always returns `null`

Reply:
It's overridden in `BoxDeploy`